### PR TITLE
257 FIx Informational Issues - Logic Should be Moved To an Separate Function - Refactoring

### PR DESCRIPTION
**Pull request by b-yap**
continuation of https://github.com/pendulum-chain/spacewalk/pull/385. Please review the base PR before continuing with this.

* ## 9B2-04 | Logic Should be Moved To an Separate Function - Refactoring (partial)
  * ## **from audit:**
    * Refactoring is recommended in the function `validate_stellar_transaction()` from stellar_relay/lib.rs.
    * `run_service()` from vault/src/system.rs
  * ## new file
    * validation.rs (covers the logic of `validate_stellar_transaction()`

### CODE REVIEW

File: pallets/stellar-relay/src/validation.rs [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+	let node_id_found = validators
+		.iter()
+		.any(|validator| validator.public_key.to_vec() == node_id.to_encoding());
+
+	ensure!(node_id_found, Error::<T>::EnvelopeSignedByUnknownValidator);
+
+	Ok(node_id)
+}
+
+pub fn check_targets<T: Config>(
```
Comment (ebma) [inline pallets/stellar-relay/src/validation.rs:105]: Nit: Maybe we can give this a better name. Maybe `check_quorum_validity` or `check_for_valid_quorum_set`

File: pallets/stellar-relay/src/validation.rs [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+	let targeted_organization_count = targeted_organization_map.len();
+
+	// Check that the distinct organizations occurring in the validator structs related to
+	// the externalized messages are more than 2/3 of the total amount of organizations in
+	// the tier 1 validator set.
+	// Use multiplication to avoid floating point numbers.
+	ensure!(
+		targeted_organization_count * 3 > orgs_length * 2,
+		Error::<T>::InvalidQuorumSetNotEnoughOrganizations
+	);
```
Comment (ebma) [inline pallets/stellar-relay/src/validation.rs:74]: I would move this part also to the `check_targets()` function. It's not wrong here but it's better fitting to the other one, especially if we rename it to `check_for_valid_quorum_set` as this part of related to that.

**Review by ebma — APPROVED** [review]
These are really great changes, thanks for the great refactoring 🎉  I left two nitpicky comments but besides that, IMO it's ready to be merged into the other branch.

File: pallets/stellar-relay/src/validation.rs (hunk 2 above)
Comment (b-yap) [reply pallets/stellar-relay/src/validation.rs:74]: updated

File: pallets/stellar-relay/src/validation.rs (hunk 1 above)
Comment (b-yap) [reply pallets/stellar-relay/src/validation.rs:105]: updated

**Review by ebma — APPROVED** [review]

**Pull request merged**